msdk-rawdata/src/main/java/io/github/msdk/filtering/cropper/CropFilterMethod.java

```diff
@@ -0,0 +1,103 @@
+/* 
+ * (C) Copyright 2015 by MSDK Development Team
+ *
+ * This software is dual-licensed under either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package io.github.msdk.filtering.cropper;
+
+import com.google.common.collect.Range;
+import io.github.msdk.MSDKException;
+import io.github.msdk.MSDKMethod;
+import io.github.msdk.datamodel.datapointstore.DataPointStore;
+import io.github.msdk.datamodel.msspectra.MsSpectrumDataPointList;
+import io.github.msdk.datamodel.rawdata.MsScan;
+import io.github.msdk.datamodel.rawdata.RawDataFile;
+import io.github.msdk.datamodel.datapointstore.DataPointStoreFactory;
+import io.github.msdk.datamodel.impl.MSDKObjectBuilder;
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CropFilterMethod implements MSDKMethod<RawDataFile> {
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private final @Nonnull RawDataFile rawDataFile;
+    private final @Nonnull Range<Double> mzRange;
+    private final @Nonnull Range<Float> rtRange;
+
+    private int processedScans = 0, totalScans = 0;
+    private RawDataFile result;
+    private boolean canceled = false;
+
+    public CropFilterMethod(@Nonnull RawDataFile rawDataFile, @Nonnull Range<Double> mzRange, @Nonnull Range<Float> rtRange) {
```

Comment:
In MSDK, we use the ChromatographyInfo class for retention times. ChromatographyInfo can accommodate multiple chromatography dimensions (such as in GCxGC-MS). So your rtRange should be defined as @Nonnull Range<ChromatographyInfo> rtRange.